content/en/docs/performance/resource_consumption.md

```diff
@@ -0,0 +1,122 @@
+---
+draft: false
+linktitle: Kmesh resource consumption
+menu:
+  docs:
+    parent: performance
+    weight: 2
+title: Kmesh resource consumption
+toc: true
+type: docs
+---
+# 1. Environment setup
+![resource_env](images/resource_test_env.png)
+
+- K8S version: v1.27
+- Kmesh version: 0.4
+- Kernel version: 5.10
+- Node flavor: 8U16G
+- testing tool: fortio
+- metric collecting tools: top, bpftop
+
+
+# 2. Test case 1 - POD with CPU limit scenario，collect Kmesh eBPF CPU usage
+## 2.1 Deploy 1 App A and set App A POD's CPU limit，then generage load and collect Kmesh eBPF CPU usage
+![resource_test1](images/resource_test1.png)
+
+Set the CPU limit for App A to 1 (1 CPU), and collect corresponding Kmesh eBPF CPU consumption.
+
+**Note: With the system having 8 cores and a CPU limit of 1, it means the POD can consume up to 12.5% of the CPU.**
+
+Testing result：
+![resource_result1](images/resource_test_result1.png)
+
+**Results and conclusions: When APP A uses up 1 CPU, the user space occupies 4%, the kernel space occupies 7.4%, totaling 11.4% of CPU consumption, which is less than the POD CPU limit of 12.5%. There are two possible reasons.**
+
+- APP A and Kmesh eBPF share the POD CPU limit, with Kmesh eBPF CPU being restricted by the POD CPU limit.
+- It is possible that due to the high performance of eBPF, APP A is not generating enough load to cause eBPF to exceed the CPU limit, further experiments in section 2.2 are needed.
+
+## 2.2 Deploy multiple instances of App A, set a fixed CPU limit, generate load, and collect the corresponding Kmesh eBPF CPU consumption.
+![resource_test2](images/resource_test2.png)
+
+Start 4 instances of App A, with a CPU limit of 250m for each instance, totaling 1 CPU for all 4 instances.
+
+Testing result：
+![resource_result2](images/resource_test_result2.png)
+
+When App A consumes 1 CPU, the user space occupies 4.4%, the kernel space occupies 7.7%, totaling 12.1% CPU usage. This is less than the POD CPU limit of 12.5%.
+
+**Results and Conclusion: When App A consumes 1 CPU, the user space occupies 4.4%, the kernel space occupies 7.7%, totaling 12.1% CPU usage, which is less than the POD CPU limit of 12.5%. To further validate this conclusion, Experiment 2.3 will be conducted.**
+
+## 2.3 Based on test 2.2, modify the eBPF code to decrease its performance, causing it to consume more CPU. Observe if it can exceed the POD CPU limit.
+
+Add for loop in Kmesh eBPF code：
+```c
+for (i;0;i<10000;i++) {}
```

Comment:
Won't the compiler optimize these code away?

Reply:
According to the test there is no optimization